job/storage/postgres/postgres.go

```diff
@@ -35,7 +35,12 @@ func New(dsn string) *DB {
 
 // GetAll returns all persisted Jobs.
 func (d DB) GetAll() ([]*job.Job, error) {
-	query := fmt.Sprintf(`select coalesce(json_agg(j.job), '[]'::json) from (select * from %[1]s) as j;`, TableName)
+	query := fmt.Sprintf(`with  clean as (
+select ctid ,
+(job -> 'metadata' -> 'last_attempted_run') as last_run, row_number() over(partition by (job->'id') order by  (job -> 'metadata' -> 'last_attempted_run') desc) as num 
+from %s where CAST(job ->> 'is_done' as boolean) = false
+) 
+select  coalesce(json_agg(j.job), '[]'::json) from %s j where ctid in (select ctid from clean where num = 1);`, TableName, TableName)
```

Comment:
A quite surprising behavior, and may yield subtle bugs: "GetAll" does not actually returning "all", but only a subset from this filter. How about return deduplicated result but regardless of the job's completion status to reserve the current semantic?
```suggestion
	query := fmt.Sprintf(
		`with incomplete_jobs as (
			select ctid,
					row_number() over(partition by (job->'id') order by (job->'metadata'->'last_attempted_run') desc) as row_num 
				from %s) 
		select coalesce(json_agg(j.job), '[]'::json)
		from %s j
		where ctid in (select ctid from incomplete_jobs where row_num = 1);`,
		TableName, TableName)
```